### PR TITLE
Homework/2/dmmi

### PR DESCRIPTION
**Pull request by YoungDima**
первыйнах

### CODE REVIEW

File: homework/dmmi/index.js [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-  return this.name;
+function isPalindrome (input) {
+   const inputArray = (input !== null) ? input.toString().toLowerCase().split('') : 'null';
+   for (let i = 0; i <= inputArray.length / 2 - 1; i++) {
+      if (inputArray[i] !== inputArray[inputArray.length - 1 - i]) {
+         return false;
+      }
+   }
+   return true;
 }
```
Comment (kovm) [inline homework/dmmi/index.js]: isPalindrome() throws an exception
isPalindrome([1,2,3,2,1]) returns true
isPalindrome({toString: () => 'kek'}) returns true

Let's do isPalindrome() function safer

File: homework/dmmi/index.js [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
 
-f1inst = new f1();
+console.log('isPalindrome for null === ' + isPalindrome(null));
```
Comment (kovm) [inline homework/dmmi/index.js]: Template literals look better in such cases
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: homework/dmmi/index.js (hunk 1 above)
Comment (YoungDima) [reply homework/dmmi/index.js]: Done in 414110fc021354ca6f05353dd15eec10e1c8cad7

File: homework/dmmi/index.js (hunk 2 above)
Comment (YoungDima) [reply homework/dmmi/index.js]: Done in 414110fc021354ca6f05353dd15eec10e1c8cad7

**Pull request merged**